.github/workflows/main.yml

```diff
@@ -48,35 +48,53 @@ jobs:
           repository: "mcagov/beacons-webapp"
       - id: get_webapp_commit_hash
         name: Get commit hash from Webapp repo
+        if: github.event.inputs.updateImageTags == 'true'
         run: echo "::set-output name=webapp_commit_hash::$(git rev-parse HEAD)"
       - uses: actions/checkout@v2
         with:
           repository: "mcagov/beacons-service"
       - id: get_service_commit_hash
         name: Get commit hash from Service repo
+        if: github.event.inputs.updateImageTags == 'true'
         run: echo "::set-output name=service_commit_hash::$(git rev-parse HEAD)"
+      - uses: actions/checkout@v2
+      - uses: hashicorp/setup-terraform@v1
+      - name: Save image tags for dev environment
+        if: github.event.inputs.updateImageTags == 'true'
+        run: |
+          cd terraform
+          echo "webapp_image_tag = \"${{ steps.get_webapp_commit_hash.outputs.webapp_commit_hash }}\"" > dev.images.tfvars
+          echo "service_image_tag = \"${{ steps.get_service_commit_hash.outputs.service_commit_hash }}\"" >> dev.images.tfvars
+          terraform fmt
+          git config user.name github-actions
+          git config user.email github-actions@github.com
+          git add .
+          git commit -m "Update image tags used in dev environment [skip ci]"
```

Comment:
All the steps in this job are skipped if `updateImageTags` isn't `true`.

We need `get_and_save_docker_image_tags` to run, because `terraform_deploy_dev` depends on it.

But we don't want its steps to run when `updateImageTags` isn't `true` meaning there aren't new image tags.
- If we run it when the latest image tags haven't changed, this job will fail 
- Because the commit line will exit with an error if there's no diff to commit
- `git commit -m "Update image tags used in dev environment [skip ci]`

So we only want these steps to be run when we know there is a new latest image tag, which is when the `webapp` or `service` triggers the dispatch event with the `updateImageTags = 'true'` input.

I couldn't figure out a cleaner way of doing this. 😫

Reply:
Ah that's really frustrating. To reduce the amount of code, you could just add the `if` block on the actual committing? The pipeline isn't going to take that long, this job takes about 8s, so allowing it to run to this stage is probably okay...